include/boost/math/special_functions/prime_sieve.hpp

```diff
@@ -0,0 +1,378 @@
+// Copyright 2020 Matt Borland
+//
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MATH_SPECIAL_FUNCTIONS_PRIME_SIEVE_HPP
+#define BOOST_MATH_SPECIAL_FUNCTIONS_PRIME_SIEVE_HPP
+
+#include <boost/math/special_functions/prime.hpp>
+#include <boost/assert.hpp>
+#include <deque>
+#include <vector>
+#include <iterator>
+#include <cmath>
+#include <thread>
+
+#if __cplusplus >= 201703
+#include <execution>
+#endif
+
+namespace boost { namespace math { namespace detail
+{
+// https://mathworld.wolfram.com/SieveofEratosthenes.html
+// https://www.cs.utexas.edu/users/misra/scannedPdf.dir/linearSieve.pdf
+template<class Z, class Container>
+void linear_sieve(Z upper_bound, Container &c)
+{
+    size_t least_divisors_size{static_cast<size_t>(upper_bound + 1)};
+    Z *least_divisors{new Z[least_divisors_size]{0}};
```

Comment:
This should be a `std::unique_ptr<Z[]>`, raw new/delete should be (almost) banned ;)

There's another "naked" call to new later on that should also change IMO.

Reply:
@jzmaddock I can move that one into the 21st century. The other one calls `memset()` which is not compatible with smart pointers. Also not particularly good C++ style.

Reply:
Just call unique_ptr::get() to obtain the raw pointer and pass to memset.